### ISSUE TITLE
Jobs are failing on HPC cluster with shared file system

### ISSUE THREAD

**Issue by berguner** [DataBiosphere/toil#2795]
We are still experiencing this problem with Toil 3.20.0. We are using beeGFS shared file system for all the worker nodes and the TMPDIR is also located there. The failed jobs sometimes finish successfully in a second attempt but it is not certain that the workflow will finish successfully. Here is the error we get:

`WARNING:toil.leader:r/j/jobe3n311c3    subprocess.CalledProcessError: Command '['du', '-s', '/scratch/users/berguener/tmpdir/toil-db817b2d-5165-4aab-95f6-242a975c5904-14556eaa535a0508c8a6b3bf00000017/tmpdkbz4mcb/919d098d-9fcc-4381-9a85-5ae0a32788c9']' returned non-zero exit status 1.`

_Originally posted by @berguner in https://github.com/DataBiosphere/toil/pull/1573#issuecomment-530282853_

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-420)
┆Issue Number: TOIL-420

**Comment by berguner** [comment]
After further investigations I figured out that the problem occurs when multiple worker nodes are involved. The problem does not appear if I force the jobs to a single node. I do this by the the Slurm's  option '-x' to exclude all nodes but one. I also tried setting `--clean never --cleanWorkDir never` but it did not help. I suspect that fileStore.findAndHandleDeadJobs() is still misbehaving.

At this point, I am not sure if the shared file system is part of the problem because the error did not occur even though the leader process was on a different node from the **single** worker node.

Here is how I run the `sort.py` example:
`python sort.py --clean always --disableCaching --batchSystem Slurm --disableAutoDeployment --numLines=50000 --lineLength=10 --overwriteOutput=True --N 1000000 teststore`

And here are the 2 distinct error/warning messages I get when multiple nodes are involved:

```
WARNING:toil.leader:The job seems to have left a log file, indicating failure: 'up' 5/r/job519h2gng
WARNING:toil.leader:5/r/job519h2gng    INFO:toil.worker:---TOIL WORKER OUTPUT LOG---
WARNING:toil.leader:5/r/job519h2gng    INFO:toil:Running Toil version 3.20.0-cf34ca3416697f2abc816b2538f20ee29ba16932.
WARNING:toil.leader:5/r/job519h2gng    WARNING:toil.resource:'JTRES_e2d90badf01c3474edd49787491eb5fd' may exist, but is not yet referenced by the worker (KeyError from os.environ[]).
WARNING:toil.leader:5/r/job519h2gng    WARNING:toil.fileStore:Starting job t/E/jobh7retk82/g/tmp5yd9g9o6-_serialiseJob-stream with less than 10% of disk space remaining.
WARNING:toil.leader:5/r/job519h2gng    WARNING:toil.resource:'JTRES_e2d90badf01c3474edd49787491eb5fd' may exist, but is not yet referenced by the worker (KeyError from os.environ[]).
WARNING:toil.leader:5/r/job519h2gng    Traceback (most recent call last):
WARNING:toil.leader:5/r/job519h2gng      File "/scratch/users/berguener/pyvirtual/toil/lib/python3.6/site-packages/toil/worker.py", line 331, in workerScript
WARNING:toil.leader:5/r/job519h2gng        job._runner(jobGraph=jobGraph, jobStore=jobStore, fileStore=fileStore)
WARNING:toil.leader:5/r/job519h2gng      File "/scratch/lab_bsf/modules/applications/python/3.6.7/lib/python3.6/contextlib.py", line 88, in __exit__
WARNING:toil.leader:5/r/job519h2gng        next(self.gen)
WARNING:toil.leader:5/r/job519h2gng      File "/scratch/users/berguener/pyvirtual/toil/lib/python3.6/site-packages/toil/fileStore.py", line 1686, in open
WARNING:toil.leader:5/r/job519h2gng        jobState = self._readJobState(self.jobStateFile)
WARNING:toil.leader:5/r/job519h2gng      File "/scratch/users/berguener/pyvirtual/toil/lib/python3.6/site-packages/toil/fileStore.py", line 1848, in _readJobState
WARNING:toil.leader:5/r/job519h2gng        with open(jobStateFileName, 'rb') as fH:
WARNING:toil.leader:5/r/job519h2gng    FileNotFoundError: [Errno 2] No such file or directory: '/scratch/users/berguener/tmpdir/toil-8d7ba8be-fc3c-4cb3-80c6-fb1baee6d66d-14556eaa535a0508c8a6b3bf00000017/tmpbyh_ybi5/d9d0a8fb-b262-4143-8946-489ff9ecb80d/.jobState'
WARNING:toil.leader:5/r/job519h2gng    ERROR:toil.worker:Exiting the worker because of a failed job on host n010
WARNING:toil.leader:5/r/job519h2gng    WARNING:toil.jobGraph:Due to failure we are reducing the remaining retry count of job 'up' 5/r/job519h2gng with ID 5/r/job519h2gng to 1
WARNING:toil.leader:5/r/job519h2gng    WARNING:toil.jobGraph:We have increased the default memory of the failed job 'up' 5/r/job519h2gng to 2147483648 bytes
INFO:toil.leader:Issued job 'up' 5/r/job519h2gng with job batch system ID: 195 and cores: 1, disk: 2.0 G, and memory: 2.0 G


INFO:toil.leader:Job ended successfully: 'up' X/5/jobz9juoaoh
WARNING:toil.leader:The job seems to have left a log file, indicating failure: 'up' X/5/jobz9juoaoh
WARNING:toil.leader:X/5/jobz9juoaoh    INFO:toil.worker:---TOIL WORKER OUTPUT LOG---
WARNING:toil.leader:X/5/jobz9juoaoh    INFO:toil:Running Toil version 3.20.0-cf34ca3416697f2abc816b2538f20ee29ba16932.
WARNING:toil.leader:X/5/jobz9juoaoh    WARNING:toil.resource:'JTRES_e2d90badf01c3474edd49787491eb5fd' may exist, but is not yet referenced by the worker (KeyError from os.environ[]).
WARNING:toil.leader:X/5/jobz9juoaoh    WARNING:toil.fileStore:Starting job x/7/jobk1c7ftyy/g/tmpqgmzyn8x-_serialiseJob-stream with less than 10% of disk space remaining.
WARNING:toil.leader:X/5/jobz9juoaoh    WARNING:toil.resource:'JTRES_e2d90badf01c3474edd49787491eb5fd' may exist, but is not yet referenced by the worker (KeyError from os.environ[]).
WARNING:toil.leader:X/5/jobz9juoaoh    Traceback (most recent call last):
WARNING:toil.leader:X/5/jobz9juoaoh      File "/scratch/users/berguener/pyvirtual/toil/lib/python3.6/site-packages/toil/worker.py", line 331, in workerScript
WARNING:toil.leader:X/5/jobz9juoaoh        job._runner(jobGraph=jobGraph, jobStore=jobStore, fileStore=fileStore)
WARNING:toil.leader:X/5/jobz9juoaoh      File "/scratch/lab_bsf/modules/applications/python/3.6.7/lib/python3.6/contextlib.py", line 88, in __exit__
WARNING:toil.leader:X/5/jobz9juoaoh        next(self.gen)
WARNING:toil.leader:X/5/jobz9juoaoh      File "/scratch/users/berguener/pyvirtual/toil/lib/python3.6/site-packages/toil/fileStore.py", line 1686, in open
WARNING:toil.leader:X/5/jobz9juoaoh        jobState = self._readJobState(self.jobStateFile)
WARNING:toil.leader:X/5/jobz9juoaoh      File "/scratch/users/berguener/pyvirtual/toil/lib/python3.6/site-packages/toil/fileStore.py", line 1848, in _readJobState
WARNING:toil.leader:X/5/jobz9juoaoh        with open(jobStateFileName, 'rb') as fH:
WARNING:toil.leader:X/5/jobz9juoaoh    FileNotFoundError: [Errno 2] No such file or directory: '/scratch/users/berguener/tmpdir/toil-8d7ba8be-fc3c-4cb3-80c6-fb1baee6d66d-14556eaa535a0508c8a6b3bf00000017/tmpxjhhtfzr/2720020d-f9e5-4010-9909-55a542b9bb68/.jobState'
WARNING:toil.leader:X/5/jobz9juoaoh    ERROR:toil.worker:Exiting the worker because of a failed job on host n010
WARNING:toil.leader:X/5/jobz9juoaoh    WARNING:toil.jobGraph:Due to failure we are reducing the remaining retry count of job 'up' X/5/jobz9juoaoh with ID X/5/jobz9juoaoh to 1
WARNING:toil.leader:X/5/jobz9juoaoh    WARNING:toil.jobGraph:We have increased the default memory of the failed job 'up' X/5/jobz9juoaoh to 2147483648 bytes
INFO:toil.leader:Issued job 'up' X/5/jobz9juoaoh with job batch system ID: 196 and cores: 1, disk: 2.0 G, and memory: 2.0 G
```

**Comment by ionox0** [comment]
I'm not sure if this is the issue you're having, but can you check the the worker log files if you used the `--logDir` parameter, and see if there wasn't a different underlying error?

I feel that I may have once experienced this issue while Toil is trying to cleanup a job that failed for another reason

You may be able to use `--restart` and `--logDebug` if you didn't save the worker logs and haven't deleted this run.

**Comment by berguner** [comment]
I am running sort.py to test my toil installation, so reproducing the issue is not a problem.

I tried to use `--logDir` but it says there is no such parameter.

**Comment by adamnovak** [comment]
We think this may have been fixed with #4440 or other changes recently to better tolerate inconsistent filesystems. We're going to make a release today; please let us know if this problem is still reproducible in that release.
